unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -4359,22 +4359,31 @@ float allowedError
                 return;
             }
 
-            var metadata = assetDb.prefabs.Where(p => p.GetComponent<SimObjPhysics>() != null).Select(
-                p => {
-                    var simObj = p.GetComponent<SimObjPhysics>();
-                    var bb = simObj.AxisAlignedBoundingBox;
-                    return new AssetMetadata() {
-                        id = simObj.gameObject.name,
-                        type = simObj.Type.ToString(),
-                        primaryProperty = simObj.PrimaryProperty.ToString(),
-                        secondaryProperties = simObj.SecondaryProperties.Select(s => s.ToString()).ToList(),
-                        boundingBox = new BoundingBox() {
-                            min = bb.center - bb.size / 2.0f,
-                            max = bb.center + bb.size / 2.0f
-                        }
-                    };
+            var metadata = new Dictionary<string, Dictionary<string, object>>();
+            foreach (GameObject p in assetDb.prefabs) {
+                if (p.GetComponent<SimObjPhysics>() == null) {
+                    continue;
+                }
+
+                var simObj = p.GetComponent<SimObjPhysics>();
+                var bb = simObj.AxisAlignedBoundingBox;
+
+                // TODO: no objects should have duplicate names.
+                if (metadata.ContainsKey(simObj.gameObject.name)) {
+                    continue;
```

Comment:
silent error if there are duplicates

Reply:
Yes. I believe @elimvb is working on removing the duplicates. I don't want the action to fail in the meantime though.

Reply:
Makes sense